source/Calamari.Tests/Fixtures/PackageRetention/LeastFrequentlyUsedWithAgingCacheAlgorithmFixture.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Deployment.PackageRetention.Caching;
+using Calamari.Deployment.PackageRetention.Model;
+using FluentAssertions;
+using NUnit.Framework;
+
+namespace Calamari.Tests.Fixtures.PackageRetention
+{
+    [TestFixture]
+    public class LeastFrequentlyUsedWithAgingCacheAlgorithmFixture
+    {
+        [Test]
+        public void WhenSpaceNeededIsGreaterThanSpaceUsed_ThenThrowException()
+        {
+            var lfu = new LeastFrequentlyUsedWithAgingCacheAlgorithm();
+            Assert.Throws<InsufficientCacheSpaceException>(() => lfu.GetPackagesToRemove(new List<JournalEntry>(), 10_000));
+        }
+
+        [Test]
+        public void WhenUsingAllThreeFactors_ThenReturnThePackageOneWithTheLowestValue()
+        {
+            var spaceNeeded = 10;
+
+            var entries = new[]
+            {
+                CreateEntry("package-1", "1.0", 10, ("task-1", 2), ("task-2", 3)),
+                CreateEntry("package-2", "1.1", 10, ("task-3", 1)),
+                CreateEntry("package-3", "1.0", 10, ("task-4", 2)),
+                CreateEntry("package-3", "1.1", 10, ("task-5", 3))
+            };
+
+            var packagesToRemove = new LeastFrequentlyUsedWithAgingCacheAlgorithm(0.5M, 1, 1)
+                                   .GetPackagesToRemove(entries, spaceNeeded)
+                                   .OrderBy(p => p.PackageId.Value)
+                                   .ThenBy(p => p.Version)
+                                   .ToList();
+
+            packagesToRemove.Should()
+                            .ContainSingle();
+
+            var package = packagesToRemove.FirstOrDefault();
+            package.PackageId.Value.Should().BeEquivalentTo("package-3");
+            package.Version.ToString().Should().BeEquivalentTo("1.0");
+        }
+
+        [TestCaseSource(nameof(ExpectMultiplePackageIdsTestCaseSource))]
+        public void ExpectingMultiplePackages(JournalEntry[] entries, int spaceNeeded, ExpectedPackageIdAndVersion[] expectedPackageIdVersionPairs)
+        {
+            var packagesToRemove = new LeastFrequentlyUsedWithAgingCacheAlgorithm()
+                                   .GetPackagesToRemove(entries, spaceNeeded)
+                                   .OrderBy(p => p.PackageId.Value)
+                                   .ThenBy(p => p.Version)
+                                   .ToList();
+
+            packagesToRemove.Should()
+                            .SatisfyRespectively(expectedPackageIdVersionPairs
+                                                 .Select<ExpectedPackageIdAndVersion, Action<PackageIdentity>>(p =>
```

Comment:
Here we're creating actions to check the packages to remove against the expected ones.  The actions will be called by the `SatisfyRespectively(..)` method.